pkg/common/utils/sbom_utils/sbom.go

```diff
@@ -0,0 +1,29 @@
+package sbom_utils
+
+import (
+	"github.com/google/osv-scanner/pkg/lockfile"
+	"github.com/package-url/packageurl-go"
+)
+
+// ParsePurlType parses a package URL (purl) type and returns the corresponding lockfile.Ecosystem.
+// It also returns a boolean indicating whether the provided purl type is recognized.
+// The recognized purl types are mapped to their corresponding lockfile ecosystems.
+func ParsePurlType(purl_type string) (lockfile.Ecosystem, bool) {
```

Comment:
Can we rename this to something very explicit such as `ParsePurlTypeToLockfile` and return `error` in case of failure instead of `bool`. Error helps us carry the cause of failure along the control flow and is easier to debug with logs. Example error when known type is not available:

```
fmt.Errorf("no known package url type found for: %s", purl_type)
```